content/en/tutorials/google-ads-for-games.md

```diff
@@ -0,0 +1,388 @@
+
+<iframe src="https://playcanv.as/e/p/OkynewOO/" title="Finished Tutorial"></iframe>
+
+## Setting up
+
+To use Google H5 Games Ads, you will need to create a [Google Adsense account][google-adsense] which will give you a [Publisher ID][google-publisher-id] for the ad integration.
+
+You will also need to review the Google H5 Games Ads [policies][google-afg-policies] and sign up to the [beta here][google-afg-signup].
+
+Go to the [Google H5 Ad Tutorial (Start)][tutorial-template-start] project dashboard page and click on 'fork' and let's take a look at the project.
+
+This is a simple project where it's ready for us to add the button click callbacks to show the ads in the `ui-controller.js` script:
+
+```
+// ...
+// initialize code called once per entity
+UiController.prototype.initialize = function() {
+    this.rewardedGivenPanelEntity.enabled = false;
+    this.rewardedAdButtonEntity.button.active = false;
+
+    var onRefresh = function () {
+
+    }.bind(this);
+
+    this.refreshButtonEntity.button.on('click', function (e) {
+        onRefresh();
+    }, this);
+
+    this.interstitialAdButtonEntity.button.on('click', function(e) {
+
+    }, this);
+
+    this.rewardedAdButtonEntity.button.on('click', function (e) {
+
+    }, this);
+
+    this.okRewardButtonEntity.button.on('click', function (e) {
+        this.rewardedGivenPanelEntity.enabled = false;
+    }, this);
+};
+```
+
+The first step is to add the Google SDK integration files which we can do by copying the folder 'Adsense for Games' from the [Flappy Bird project][flappy-bird-ads-demo].
+
+![][copy-afg-folder]
+
+And pasting into our fork of the tutorial project.
+
+![][paste-afg-folder]
+
+Let's take a look inside:
+
+[`afg-intergration.js`][afg-intergration.js] adds Google's SDK to the page and also sets the time between ads being served to the user which we can change if need be. For the tutorial, we have also enabled test mode to show placeholders instead of real ads.
+
+`ca-pub-XXXXXXXXX` will need to be replaced with your own [Publisher ID][google-publisher-id] from Google Adsense.
+
+```
+// https://developers.google.com/ad-placement/docs/example
+(function() {
+    var script = document.createElement('script');
+    script.setAttribute('data-ad-client', 'ca-pub-XXXXXXXXX');
+
+    // Comment out to disable test mode
+    script.setAttribute('data-adbreak-test', 'on');
+
+    // Ad frequency is user configuble
+    // https://developers.google.com/ad-placement/docs/ad-rate
+    script.setAttribute('data-ad-frequency-hint', '30s');
+
+    // Comment in if you would like to use a specific channel for ad reporting
+    // https://developers.google.com/ad-placement/docs/advanced-reporting
+    // script.setAttribute('data-ad-channel', 'XXXXXX');
+
+    script.src = 'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js';
+    document.head.appendChild(script);
```

Comment:
Please update this section. We know use slightly different flavor of adding adsbygoogle.js. It's been used by the AdSense for a while (6+ months) but only know we updated ad-placement docs to reflect it. You can see it here: https://developers.google.com/ad-placement/docs/example#indexhtml_web

The main benefit is that client is now added in adsbygoogle.js url itself so we can faster load ad-placement API. Which is important for preroll ads.

Reply:
Sorry, what would you need it changed to? We can't embed script tags in the html due to the way the Editor projects work if that's what is being suggested?

This function is called first thing in the project, before the engine load and even before the assets are loaded.

Reply:
Oh wait, is it just the url and format that needs to be changed here?

Reply:
Yes. I meant you should add `adsbygoogle.js?client=ca-pub-XXXXXXXXX`, remove `data-ad-client` attribute and add `crossorigin=anonymous` attribute on the constructed script tag.